### DISCUSSION TITLE
How can i turn tauri://localhost into myapp://localhost?

### DISCUSSION THREAD

**Discussion by mq1** [tauri-apps/tauri#5944] [Q&A]
Currently I use oauth2 in my app, and my redirect uri is "tauri://localhost/login".

Is there an option to change it to "myappname://localhost/login" and make tauri work with it?

**Comment by mail4metablocks** [comment]


**Reply by mq1** [reply]
> Yes, it is possible to change the redirect URI to "myappname://localhost/login" and make Tauri work with it. Here are the steps you can follow:
> 
> ```
> Modify the redirect_uri parameter in your OAuth2 client to use the new URI scheme, "myappname://localhost/login". This will cause the OAuth2 server to redirect to this URI after the user grants permission.
> 
> In your Tauri app, you will need to register the custom URI scheme "myappname" as a valid protocol. To do this, you can use the registerProtocol method in the Tauri API. For example:
> ```
> 
> tauri.registerProtocol("myappname", (event) => { // This callback function will be called when the app is launched with a "myappname://" URI });
> 
> ```
> In the callback function, you can then parse the URI and extract the necessary information (e.g. the authorization code) from the query string. You can then use this information to complete the OAuth2 flow and obtain the desired access token.
> ```

Thanks for the answer, I'm calling register_uri_scheme_protocol from rust, but I don't really know what to put as the callback function
